core/src/core/tests/test_tools.py

```diff
@@ -0,0 +1,204 @@
+import pytest
+from ..api.memgraph import MemgraphClient
+from ..tools.schema import ShowSchemaInfoTool
+from ..tools.config import ShowConfigTool
+from ..tools.index import ShowIndexInfoTool
+from ..tools.storage import ShowStorageInfoTool
+from ..tools.constraint import ShowConstraintInfoTool
+from ..tools.trigger import ShowTriggersTool
+from ..tools.page_rank import PageRankTool
+from ..tools.cypher import CypherTool
+from ..tools.betweenness_centrality import BetweennessCentralityTool
+from ..utils.logging import logger_init
+
+logger = logger_init("test-tools")
+
+
+def test_show_schema_info_tool():
+    """Test the ShowSchemaInfo tool."""
+
+    uri = "bolt://localhost:7687"
+    user = "memgraph"
+    password = "memgraph"
+
+    memgraph_client = MemgraphClient(uri=uri, username=user, password=password)
+
+    schema_tool = ShowSchemaInfoTool(db=memgraph_client)
+    assert "show_schema_info" in schema_tool.name
+
+    result = schema_tool.call({})
+
+    assert isinstance(result, list)
+    assert len(result) >= 1
+    schema_tool.close()
```

Comment:
Going back to comment on betweenness centrality tool, it seems weird to "close" a tool. I think this should have been handled in call() for each tool, but maybe I am missing smth.

Reply:
Good catch, not needed currently.